Dispatcher/UI/Screens/AuthPage/LoginPageViewModel.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  AuthPageViewModel.swift
+//  Dispatcher
+//
+//  Created by Hadar Basson on 30/08/2023.
+//
+
+import Foundation
+
+class LoginPageViewModel: ObservableObject, AuthViewModelProtocol {
+    
+    @Published var isSecure: [String : Bool] = [Strings.passwordPlaceholder: true]
+    
+    @Published var fields: [String: String] = [
+        Strings.emailPlaceholder: "",
+        Strings.passwordPlaceholder: "",
+    ]
+
+    let inputPlaceholders = [Strings.emailPlaceholder, Strings.passwordPlaceholder]
+    
+    let buttonText = [Strings.logInButton, Strings.signUpButton]
+
+    let title = Strings.titleLoginScreen
+    
+
+    private let authRepository: AuthRepositoryProtocol
+    
+    init(authRepository: AuthRepositoryProtocol) {
+        self.authRepository = authRepository
+    }
+    
+    
+    func setValue(key: String, value: String) {
+        fields[key] = value
+    }
+    
+    func authentication(doWhenFinish:(()->Void)) {
+        if(isValid()) {
+            authRepository.login(email: fields[Strings.emailPlaceholder]!, password: fields[Strings.passwordPlaceholder]!)
+            doWhenFinish()
```

Comment:
why do we pass a callback? just return a Bool and execute it where it was passed